### DISCUSSION TITLE
How to override built-in darkVariants?

### DISCUSSION THREAD

**Discussion by lianghx-319** [tailwindlabs/tailwindcss#6099] [Help]
I want to i integration third-party UI library and tailwindcss. And make a same dark mode selector. Either `class` `.dark` or `media`

**Comment by lianghx-319** [comment]
Anyone can help! 🙏

**Comment by lianghx-319** [comment]
https://github.com/tailwindlabs/tailwindcss/blob/f94a117500266a8f5e9cfec3b03db81493b1b047/src/lib/setupContextUtils.js#L568-L580

It seems work moving `darkVariants ` from `afterVariants` to `beforeVariants`, so we can use a custom variant to override the built-in dark variant?

**Comment by pzuraq** [comment]
I'm encountering the same issue here, we'd like to override this to match our theming setup. Is there any particular reason that `darkVariants` comes after user plugins?

**Comment by lunatiqqc** [comment]
I would like to change the name of the non configurable "dark:" prefix, to for example "light:" for a dark-mode first design

**Reply by mybearworld** [reply]
You can do this by creating a custom plugin: https://play.tailwindcss.com/EZmqlziwt5
